cli/src/commands/extensions/api.rs

```diff
@@ -0,0 +1,175 @@
+//! Most functions in this module are marked `#[allow(unused)]` for the time being. This
+//! intentional, as there are no clients for those functions in the rest of the code, but
+//! those functions will have to be used by the Deno integration; at that point, we may
+//! remove the annotations.
+
+use std::path::Path;
+use std::str::FromStr;
+
+use crate::commands::parse::{get_packages_from_lockfile, LOCKFILE_PARSERS};
+use crate::config::{get_current_project, Config};
+use crate::{api::PhylumApi, auth::UserInfo};
+
+use anyhow::{anyhow, Context, Error, Result};
+use deno_core::OpState;
+use once_cell::sync::Lazy;
+use phylum_types::types::auth::{AccessToken, RefreshToken};
+use phylum_types::types::common::{JobId, ProjectId};
+use phylum_types::types::job::JobStatusResponse;
+use phylum_types::types::package::{
+    Package, PackageDescriptor, PackageStatusExtended, PackageType,
+};
+use phylum_types::types::project::ProjectDetailsResponse;
+
+/// Container for lazily evaluated dependencies of Extensions API functions. These values won't be
+/// visible from extension code and will have to be set up by the JS runtime builder which will
+/// load their configuration as any other command and then provide the pertinent factories.
+struct InjectedDependencies {
+    api: Lazy<PhylumApi>,
+    config: Lazy<Config>,
+}
+
+impl InjectedDependencies {
+    #[allow(unused)]
+    pub(super) fn from_factories(
+        api_factory: fn() -> PhylumApi,
+        config_factory: fn() -> Config,
+    ) -> Self {
+        InjectedDependencies {
+            api: Lazy::new(api_factory),
+            config: Lazy::new(config_factory),
+        }
+    }
+}
+
+#[allow(unused)]
+pub(super) async fn phylum_analyze(
+    state: &mut OpState,
+    lockfile: &str,
+    project: Option<&str>,
+    group: Option<&str>,
+) -> Result<ProjectId> {
+    let api = &mut state.borrow_mut::<InjectedDependencies>().api;
+    let (packages, request_type) = get_packages_from_lockfile(Path::new(lockfile))
+        .context("Unable to locate any valid package in package lockfile")?;
+
+    let (project, group) = match (project, group) {
+        (Some(project), group) => (api.get_project_id(project, group).await?, None),
+        (None, _) => {
+            if let Some(p) = get_current_project() {
+                (p.id, p.group_name)
+            } else {
+                return Err(anyhow!("Failed to find a valid project configuration"));
+            }
+        }
+    };
+
+    let job_id = api
+        .submit_request(
+            &request_type,
+            &packages,
+            false,
+            project,
+            None,
+            group.map(String::from),
+        )
+        .await?;
+
+    Ok(job_id)
```

Comment:
It is an actionable value, meaning it could be an input to other calls (e.g. suppose the extension wants to implement a polling mechanism to check for status).

Reply:
Is there a way to query for the status with this API though? Otherwise the use of this function just seems somewhat limited. The ID by itself is useless unless you have something you can do with it.

Reply:
We should probably add that. As I mentioned on Notion, the API surface is a little limited as it is, only exposing equivalent CLI endpoints. I suggest that once the general scaffolding for running an extension which invokes API functions is in place, we start expanding the API surface, but we should have a clearer idea about how to pass data around first.

Reply:
The CLI's analyze command right now does more than just return the Job ID though, right? So I think when trying to appropriately mimic its API surface, we should return the result of the analysis?